src/problem.jl

```diff
@@ -122,9 +122,17 @@ end
 # when setindex! is used.
 function Base.setindex!(prob::JumpProblem, args...; kwargs...)
     SciMLBase.___internal_setindex!(prob.prob, args...; kwargs...)
+end
+
+# for updating parameters in JumpProblems to update MassActionJumps
+function SII.set_parameter!(prob::JumpProblem, val, idx)
+    ans = SII.set_parameter!(SII.parameter_values(prob), val, idx)
+
     if using_params(prob.massaction_jump)
         update_parameters!(prob.massaction_jump, prob.prob.p)
     end
+
+    ans
```

Comment:
While this is technically correct, `set_parameter!` will be called once for each parameter which would make this _very_ expensive. I'll PR to SII to add a callback that runs at the end of `setp` (`finalize_parameters_hook!(prob, ps)`?)

Reply:
https://github.com/SciML/SymbolicIndexingInterface.jl/pull/58

Reply:
@AayushSabharwal please ping me when a SII release with that callback is available so I can get this finished -- we need this for making updates to Catalyst for MTK9. Thanks!

Reply:
It's been released now @isaacsas

Reply:
Oops, I thought it was merged. Sorry 😅

Reply:
_Now_ it's out in 0.3.13